system/lib/libc/musl/src/unistd/dup2.c

```diff
@@ -10,7 +13,11 @@ int dup2(int old, int new)
 	while ((r=__syscall(SYS_dup2, old, new))==-EBUSY);
 #else
 	if (old==new) {
+#if __EMSCRIPTEN__
 		r = __syscall(SYS_fcntl, old, F_GETFD);
```

Comment:
I wonder why these functions don't call the external `fcntl` function instead of calling the syscall directly.   They would make it interposable so that we could intercept and replace this fcntl.